cmd/object-handlers.go

```diff
@@ -1617,6 +1617,59 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 	// Ensure that metadata does not contain sensitive information
 	crypto.RemoveSensitiveEntries(metadata)
 
+	var hasLifecycleConfig bool
+	if _, err := globalBucketMetadataSys.GetLifecycleConfig(bucket); err == nil {
+		hasLifecycleConfig = true
+	}
+	if hasLifecycleConfig {
+		// Get appropriate object info to identify the remote object to delete
+		// in the case of a bucket with versioning suspended or disabled.
+		pOpts, err := putOpts(ctx, r, bucket, object, metadata)
+		if err != nil {
+			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
+			return
+		}
+		if pOpts.VersionSuspended && pOpts.VersionID == "" {
+			pOpts.VersionID = nullVersionID
+		}
```

Comment:
putOpts is not required for GetObjectInfo call
you can directly check globalBucketVersioningSys.Suspended(bucket) and use opts.VersionID (opts already initialized ahead)